### PR TITLE
Validate consumable use

### LINKED ISSUE #145: validate consumable in action
validate consumable can use in actions(HackAndSlash, RankingBattle)

### PR DESCRIPTION
**Pull request by ipdae**
- resolve #145

### CODE REVIEW

**Comment by riemannulus** [comment]
Seems conflict 😢

**Comment by ipdae** [comment]
@riemannulus resolve conflict

**Review by boscohyun — APPROVED** [review]

File: Lib9c/GameConfig.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ public static class GameConfig
 #endif
 
         public const int SlotCount = 3;
+        public const int ConsumableSlotCount = 5;
```
Comment (longfin) [inline Lib9c/GameConfig.cs:12]: Is this constant needed?

**Review by longfin — APPROVED** [review]

File: Lib9c/GameConfig.cs (hunk 1 above)
Comment (ipdae) [reply Lib9c/GameConfig.cs:12]: yes.

Comment (longfin) [reply Lib9c/GameConfig.cs:12]: I guess it's only used in [test code] (https://github.com/planetarium/lib9c/pull/148/files#diff-bce870726bc5fa621b60324c6e112f4fb5f0d9ec21c9b9a3176b6e43cc77593bR155)... so it doesn't seem for Lib9c project right now.

Comment (ipdae) [reply Lib9c/GameConfig.cs:12]: I added it to limit the number of consumable, but there is no using in `ValidateConsumable` right now. delete this.

File: Lib9c/Model/State/AvatarState.cs [hunk 2]
```diff
@@ -405,6 +405,53 @@ public void ValidateEquipments(List<Guid> equipmentIds, long blockIndex)
...
+
+                var equipment = (Consumable) outNonFungibleItem;
+                if (equipment.RequiredBlockIndex > currentBlockIndex)
+                {
+                    throw new RequiredBlockIndexException(
+                        $"{equipment.ItemSubType} / unlock on {equipment.RequiredBlockIndex}");
+                }
+
+                int requiredLevel;
+                switch (i)
```
Comment (riemannulus) [inline Lib9c/Model/State/AvatarState.cs:427]: IMO using `slot` instead `i` is more semantic.

**Review by longfin — APPROVED** [review]

**Pull request merged**